mev_inspect/classifiers/specs/zero_ex.py

```diff
@@ -1,9 +1,51 @@
+from typing import Optional, List
+from mev_inspect.schemas.transfers import Transfer
+from mev_inspect.schemas.swaps import Swap
 from mev_inspect.schemas.traces import (
+    DecodedCallTrace,
     Protocol,
 )
 from mev_inspect.schemas.classifiers import (
     ClassifierSpec,
+    SwapClassifier,
 )
+from mev_inspect.classifiers.helpers import (
+    is_valid_0x_swap,
+    get_0x_token_in_data,
+    get_0x_token_out_data,
+)
+
+
+class ZeroExSwapClassifier(SwapClassifier):
+    @staticmethod
+    def parse_swap(
+        trace: DecodedCallTrace,
+        prior_transfers: List[Transfer],
+        child_transfers: List[Transfer],
+    ) -> Optional[Swap]:
+
+        assert is_valid_0x_swap(trace, child_transfers)
+
+        token_in_address, token_in_amount = get_0x_token_in_data(trace, child_transfers)
+
+        token_out_address, token_out_amount = get_0x_token_out_data(trace)
```

Comment:
love the breakdown into helper functions ❤️ 

those should live as private functions in here rather than in the "shared" helpers file